presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcResultSetTimezone.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.jdbc;
+
+import io.airlift.log.Logging;
+import io.prestosql.server.testing.TestingPrestoServer;
+import org.testcontainers.containers.OracleContainer;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableList;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimeZone;
+
+import static io.prestosql.jdbc.TestPrestoDriver.closeQuietly;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Checks that JDBC dates and timestamps are interpreted in the client JVM timezone and should not
+ * be affected by the session timezone.  We also compare the behavior to a reference JDBC driver
+ * (we picked Oracle).
+ */
+@Test(singleThreaded = true)
+public class TestJdbcResultSetTimezone
+{
+    private TestingPrestoServer server;
+    private List<ReferenceDriver> referenceDrivers;
+
+    private Connection connection;
+    private Statement statement;
+
+    @BeforeClass
+    public void setupServer()
+    {
+        Logging.initialize();
+        server = TestingPrestoServer.create();
+        referenceDrivers = ImmutableList.of(new OracleReferenceDriver());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDownServer()
+    {
+        closeQuietly(server);
+        referenceDrivers.forEach(ReferenceDriver::close);
```

Comment:
If one of these throws, the rest won't be closed. A better way is to use a `Closer`.

Also, we avoid using `forEach()` for simple for-each loops. Only use it when there is a specific benefit, such as the terminal operation of a stream.